### PR TITLE
SALTO-5271: Support label attribute

### PR DESCRIPTION
**Pull request by idoamit8**
In this PR:
1. Added CV that prevents the deletion of label attribute.
2. Added a filter that fetches and deploys the label attribute.
3. Modified the attribute deploy filter. 

---

_Additional context for reviewer_
None

---
_Release Notes_: 
_Jira Adapter_:
1. New instance element that holds for each objectType who its label attribute. 
2. New CV that prevents deletion of assets label attribute. 

---
_User Notifications_: 
_Jira Adapter_:
1. New instance element that holds for each objectType who its label attribute. 
2. New CV that prevents deletion of assets label attribute.

### CODE REVIEW

File: packages/jira-adapter/src/filters/assets/label_object_type_attribute.ts [hunk 1]
```diff
@@ -0,0 +1,161 @@
...
+        attributes,
+        labelAttributeType,
+        instanceNameToInstacne[parentName]
+      )
+      if (labelAttributeInstance === undefined) {
+        return
+      }
+      elements.push(labelAttributeInstance)
+    })
+    // Remove label field field from attributes
```
Comment (omrilit) [inline packages/jira-adapter/src/filters/assets/label_object_type_attribute.ts:105]:
```suggestion
    // Remove label value from attributes
```

Comment (omrilit) [inline packages/jira-adapter/src/filters/assets/label_object_type_attribute.ts:105]: Should we also remove the label field from the Attribute Type?

File: packages/jira-adapter/src/filters/assets/label_object_type_attribute.ts [hunk 2]
```diff
@@ -0,0 +1,161 @@
...
+
+    const [relevantChanges, leftoverChanges] = _.partition(
+      changes,
+      change => getChangeData(change).elemID.typeName === OBJECT_TYPE_LABEL_ATTRIBUTE_TYPE
+    )
+    const workspaceId = await getWorkspaceId(client)
+    if (workspaceId === undefined) {
+      log.error(`Skip deployment of ${OBJECT_TYPE_LABEL_ATTRIBUTE_TYPE} types because workspaceId is undefined`)
+      const errors = relevantChanges.map(change => createSaltoElementError({
+        message: `The following changes were not deployed, due to error with the workspaceId: ${relevantChanges.map(c => getChangeData(c).elemID.getFullName()).join(', ')}`,
```
Comment (omrilit) [inline packages/jira-adapter/src/filters/assets/label_object_type_attribute.ts:127]: Probably should be reviewed by Lior

File: packages/jira-adapter/src/filters/assets/label_object_type_attribute.ts [hunk 3]
```diff
@@ -0,0 +1,161 @@
...
+      }))
+      return {
+        deployResult: { appliedChanges: [], errors },
+        leftoverChanges,
+      }
+    }
+    const deployResult = await deployChanges(
+      relevantChanges.filter(isInstanceChange),
+      async change => {
+        if (isRemovalChange(change)) {
```
Comment (omrilit) [inline packages/jira-adapter/src/filters/assets/label_object_type_attribute.ts:139]: Could you please add a comment explaining that this is covered in the CV?
I wonder whether the CV relevant code should be written here as well (make sure that if the removal is allowed, there is also a removal of its parent Object Type), WDYT?

File: packages/jira-adapter/src/filters/assets/label_object_type_attribute.ts [hunk 4]
```diff
@@ -0,0 +1,161 @@
...
+          return
+        }
+        const instance = getChangeData(change)
+        const attribute = isReferenceExpression(instance.value.labelAttribute)
+          ? instance.value.labelAttribute.value : undefined
+        if (attribute === undefined) {
+          return
+        }
+        await client.put({
```
Comment (omrilit) [inline packages/jira-adapter/src/filters/assets/label_object_type_attribute.ts:148]: Is there a need to update also the other attribute that was previously marked as label?

**Review by omrilit — COMMENTED** [review]
Looks great!
Could you please reason why you chose to model the label attribute as a dedicated instance and not just as a value on the attribute instance?

File: packages/jira-adapter/src/filters/assets/label_object_type_attribute.ts (hunk 4 above)
Comment (idoamit8) [reply packages/jira-adapter/src/filters/assets/label_object_type_attribute.ts:148]: No, it takes the label from it. (Automatically updated)

**Pull request merged**